app/src/main/java/com/android/calendar/Event.java

```diff
@@ -376,6 +399,56 @@ private static Event generateEventFromCursor(Cursor cEvents, Context context) {
         return e;
     }
 
+    /** Android's RRULE code is broken in a way the creates additional events in certain
+     *  circumstances (though never doesn't create the actaul event) so let's use another RRULE
+     *  parser to validate if the event is real or not.
+     *
+     *  In this case we're using lib-recur from https://github.com/dmfs/lib-recur through maven.
+     *
+     **/
+    static int checkRRuleEventDate( String rrule, long startTime, int endDay) {
+        // Convert the startTime into some useable Day/Month/Year values.
+        Date date = new java.util.Date(startTime);
+
+        // We'll use SimpleDateFormat to get the D/M/Y but we also need to set the timezone.
+        SimpleDateFormat sdf = new java.text.SimpleDateFormat();
+        sdf.setTimeZone(java.util.TimeZone.getTimeZone("GMT"));
+
+        sdf.applyPattern("yyyy");
+        int startYear = Integer.parseInt(sdf.format(date));
+        sdf.applyPattern("MM");
+        int startMonth = Integer.parseInt(sdf.format(date)) - 1;
+        sdf.applyPattern("dd");
+        int startDay = Integer.parseInt(sdf.format(date));
```

Comment:
Wouldn't it be preferrable to use `java.time.LocalDateTime` instead and work with epoch, since you have no associated timezone?

Untested and uncompiled:

```java
long firstInstance = LocalDateTime
    .ofEpochSecond(startTime / 1000, (startTime % 1000) * 1000000)
    .toLocalDate() // clamping to day start
    .toInstant()
    .toEpochMillis();

// comparison:
if (instance.getTimestamp() != firstInstance) {
   return 0;
} // etc.
```

Reply:
It needs to resolve out to a ```DateTime``` object to pass into ```RecurrenceSet()```, so I could work with epoch's in some of the code, but in the end it still needs to be a ```DateTime``` object, so no real point in doing both.